docs/aws/audit/elbmonitoring/rules/elb_web_tier_security_policy_remediation.mdx

```diff
@@ -1,6 +1,187 @@
+               )
+               print(f"Updated listener on port {listener['Listener']['LoadBalancerPort']} to use ELBSecurityPolicy-2016-08")
+
+   print("SSL policies updated successfully.")
+   ```
+
+4. **Run the Script**:
+   Execute the script in your Python environment. This script will iterate through all your ELBs and update the SSL negotiation policy to the latest recommended policy for all HTTPS listeners.
+
+   ```bash
+   python update_ssl_policy.py
+   ```
+
+By following these steps, you ensure that your Web-Tier ELBs are using the latest AWS security policy for SSL negotiations, thereby enhancing the security of your load balancers.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the EC2 dashboard.
+2. In the navigation pane, under LOAD BALANCING, click on Load Balancers.
+3. Select the load balancer you want to inspect. In the bottom panel, click on the "Listeners" tab.
+4. Check the SSL Certificate column. If the column is empty or the policy is not up-to-date, then the ELB is not using the latest AWS security policy for SSL negotiations.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can do this by downloading the appropriate installer from the AWS CLI website. Once installed, you can configure it by running `aws configure` and providing your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all Load Balancers: Use the following command to list all the load balancers in your account.
+
+   ```
+   aws elbv2 describe-load-balancers
+   ```
+
+   This command will return a JSON object with details about all your load balancers.
+
+3. Check the SSL Policies: For each load balancer, check the SSL policies by using the following command:
+
+   ```
+   aws elbv2 describe-ssl-policies --names [policy-name]
+   ```
+
+   Replace `[policy-name]` with the name of the SSL policy you want to check. This command will return a JSON object with details about the specified SSL policy.
+
+4. Analyze the output: Look for the "SslProtocols" field in the output. If the policy includes outdated or insecure protocols (like SSLv3 or TLS 1.0), then the load balancer is not using the latest AWS security policy for SSL negotiations.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install the necessary AWS SDK for Python (Boto3) if you haven't done so already. You can do this by running the command `pip install boto3` in your terminal.
+
+2. Import the necessary modules and establish a session with AWS using your credentials. Here's a basic setup:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>